Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/ControllerPoseSynchronizer.cs

```diff
@@ -10,7 +10,7 @@ namespace Microsoft.MixedReality.Toolkit.Input
     /// <summary>
     /// Waits for a controller to be initialized, then synchronizes its transform position to a specified handedness.
     /// </summary>
-    public class ControllerPoseSynchronizer : InputSystemGlobalListener, IMixedRealityControllerPoseSynchronizer
+    public class ControllerPoseSynchronizer : InputSystemGlobalHandlerListener, IMixedRealityControllerPoseSynchronizer
```

Comment:
>InputSystemGlobalHandlerListener [](start = 46, length = 32)

For my own understanding, is the main value in going with InputSystemGlobalHandlerListener here (instead of just being a MonoBehavior) is how it will wait for the input system to be valid before starting up? Or is there some other aspect here of non-focus/gaze related stuff?

Reply:
3 things:
* hides waiting for input system in `Start`.
* InputSystem property was used in most of derived classes, and I didn't want to copy-paste it everywhere.
* additional utility method to register all handlers for a component via reflection (used in only one place for the moment, but it will not look nice without it)

---
In reply to: [295381267](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/4970#discussion_r295381267) [](ancestors = 295381267)

Reply:
Actually I misread the class what class the question was for..
Just first 2 reasons for this one.